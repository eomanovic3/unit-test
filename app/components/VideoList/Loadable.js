/**
 *
 * Asynchronously loads the component for VideoList
 *
 */

import loadable from 'utils/loadable';

export default loadable(() => import('./index'));
