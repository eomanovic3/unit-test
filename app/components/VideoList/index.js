/**
 *
 * VideoList
 *
 */

import React from 'react';
// import PropTypes from 'prop-types';
// import styled from 'styled-components';

import PropTypes from 'prop-types';
import VideoItem from '../VideoItem';

class VideoList extends React.PureComponent {
  render() {
    if (this.props.videos && this.props.videos.results) {
      const renderendList = this.props.videos.results ? (
        this.props.videos.results.map((video, index) => (
          <div key={index.toString()}>
            <VideoItem
              key={video.id}
              video={video}
              onVideoSelect={this.props.onVideoSelect}
            />
          </div>
        ))
      ) : (
        <div />
      );
      return <div className="ui relaxed divided list">{renderendList}</div>;
    }
    return null;
  }
}

VideoList.propTypes = {
  videos: PropTypes.object,
  onVideoSelect: PropTypes.func,
};
export default VideoList;
