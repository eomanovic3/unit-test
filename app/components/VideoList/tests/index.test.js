/**
 *
 * Tests for VideoList
 *
 * @see https://github.com/react-boilerplate/react-boilerplate/tree/master/docs/testing
 *
 */

import React from 'react';
import { render } from 'react-testing-library';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import VideoList from '../index';

Enzyme.configure({ adapter: new Adapter() });

let container;
let props;
beforeEach(() => {
  container = document.createElement('div');
  props = {
    videos: {
      results: [
        {
          original_title: 'Mad Hot Ballroom',
          release_date: '2005-05-13',
          overview: 'Overview',
          id: 14358,
          poster_path: '/cVnBICXmyqlusz1iYvjvIOHxo5U.jpg',
        },
        {
          original_title: 'To Wong Foo, Thanks for Everything! Julie Newmar',
          release_date: '1995-09-07',
          overview: 'Overview',
          id: 9090,
          poster_path: '/xIDEoG9FQGmMCh5XsbkvSuD8WrW.jpg',
        },
      ],
    },
  };
  document.body.appendChild(container);
});
afterEach(() => {
  document.body.removeChild(container);
  container = null;
});

describe('<VideoList />', () => {
  it('expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<VideoList />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<VideoList />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render with videos', () => {
    const component = shallow(<VideoList videos={props.videos} />);
    expect(component).toMatchSnapshot();
  });
});
