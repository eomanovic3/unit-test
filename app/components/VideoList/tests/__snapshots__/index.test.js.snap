// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VideoList /> should render and match the snapshot 1`] = `null`;

exports[`<VideoList /> should render with videos 1`] = `ShallowWrapper {}`;
